v7.0.3+beta.2

MoojMidge opened the pull request:
### Changed
- Function and data cache are now created per user
- Video channel no longer used as studio infolabel
- Since v7.0.3+beta.1 setting values are logged to the debug log
  - Now sensitive details (location, API key, etc.) are masked
- Deprecated plugin actions removed
- Changed caching of playlists
  - First page of playlists are only cached for 5 minutes
  - Following pages are cached for 1 hour

### Fixed
- Fix error when opening My Subscriptions after login #555
- Fix directory items being treated as short videos #556
- Fix error with Setup Wizard not loading #557
- Refresh context menu item will now actually refresh listings #558

MoojMidge committed:
Fix incorrectly resetting subscription items

- Fix #555

MoojMidge committed:
Improve robustness of strptime pre-formatting

MoojMidge committed:
Fix incorrect logic when re-raising request exceptions

- Was resulting in a more generic exception being raised

MoojMidge committed:
Fix incorrect logged in status when access tokens are revoked

MoojMidge committed:
Set default request exception hook parameters for Youtube API requests

MoojMidge committed:
Disable using video channel name as studio

- Stops Kodi unnecessarily trying to load non-existent studio logos

MoojMidge committed:
Update empty v3 response log message

MoojMidge committed:
Fix hiding directory items as short videos

- Fix #556

MoojMidge committed:
Child classes properly inherit from YouTubeRequestClient

MoojMidge committed:
Remove reference to view manager

- Fix #557

MoojMidge committed:
Prevent log leaks

MoojMidge committed:
Remove support for deprecated plugin url actions

MoojMidge committed:
Use simple script entry point

MoojMidge committed:
Create per user function and data caches

- Avoid clearing caches and resetting cache items on user change

MoojMidge committed:
Cache updates to fix #558

- Make refresh context menu item actually refresh
- First page of playlist will be cached for 5 minutes, remaining pages cached for 1 hour
- Extract urls from video description cached for 1 day, rather than 1 week
- Don't clear function cache on service restart (may cause regressions but cipher/nsig handling is not currently used)

MoojMidge committed:
Misc tidy ups

MoojMidge committed:
Update workflows, changelog and remove news from addon.xml

- Update make-release workflow
  - Changelog will be used to populate release details
  - Changelog entries to follow Github markdown formatting rules
  - addon.xml updates will preserve whitespace  - news element in addon.xml will be populated from changelog without issue references
- Updated submit-release workflow
  - addon.xml updates will preserve whitespace
  - news element in addon.xml will be populated from changelog without issue references
    - news is also used for repository PRs as part of submit-release workflow    - Github references either need to be removed or updated. Removed for simplicity

MoojMidge committed:
Version bump - v7.0.3+beta.2

MoojMidge committed:
Update addon.xml and workflows to pass addon-validations

- Was failing due to empty news element. Remove element and add in release workflows

MoojMidge committed:
Don't use label formatting for category labels

- Also ensure category label is updated when name is updated

Mo1967 commented:
@MoojMidge 
Well done,
By the way;  where are the files for installation of v7.0.3+beta.2 to Nexus/Matrix ?

MoojMidge commented:
This is just a Pull Request. Once the changes are merged, then the updated installation files will be created as part of making a new release.

Not planning any further changes to v7.0.3+beta.2, so this can be merged, however there may be some changes to the associated unofficial version to allow for customisation of the sort order display, to address #559.

If it turns out the changes for the sort order customisation are too big, may just leave that along with any other updates, for a future release.

MoojMidge committed:
Fix Context argv override uri creation

- Params passed as a parameter to constructor were not being included in generated uri

MoojMidge committed:
Allow category label to be inherited

- Must be contained in directory uri to be parsed in new Context params

MoojMidge commented:
Unofficial changes are too extensive and require further testing, merging as-is to resolve v7.0.3+beta.1 bugs.
